src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -233,6 +249,32 @@ public void AbortTestRun()
 
         #endregion
 
+        private bool TryUpdateDesignMode(string runsettingsXml, out string updatedRunSettingsXml)
+        {
+            updatedRunSettingsXml = runsettingsXml;
+            var runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(runsettingsXml);
+            if (runConfiguration.DesignModeSet || !runConfiguration.TargetFrameworkSet ||
```

Comment:
I will add comments and tests before merge.

If user is already setting `DesignMode` via runsettings or CLI args; we skip. We also skip if the target framework is not known or current run is targeted to netcoreapp (since it is a breaking change; user may be running older NET.Test.Sdk; we will remove this constraint in 15.1).